### ISSUE TITLE
Wrong Result in search query

### ISSUE THREAD

**Issue by mohitgoyal201617** [pelias/api#556]
I have  inserted some openaddress, When I search NH-11 it gives me result but when i search NH 11 it does not give me result. 
this text it present in street.

**Comment by orangejulius** [comment]
Hey @mohitgoyal201617,
Thanks for reporting this! We definitely have some areas to improve around parsing text with separating characters like `-`. I think we may have some work being done now that will help, but it's generally something we are getting better at over time.

To help us out, can you send along some _full_ queries that demonstrate both the __good__ and __bad__ behavior? You can use our [compare](http://pelias.github.io/compare/#/v1/search%3Ftext=london,%20uk) tool to look at the queries, and send along the links to that tool. Thanks!

**Comment by mohitgoyal201617** [comment]
I have modified punctutions.js(Schema) commented '-' in allowed character. 
I have cloned pelias-api started it using npm start. so just want to know how to deploy schema module which is cloned and modified.

**Comment by orangejulius** [comment]
Cool, let us know how that works. I can't remember if we've tried something similar.

From the pelias/schema directory, you can run this to reset the schema:

**WARNING: ** This will remove all the data you've imported into the pelias index in Elasticsearch, and then you'll have to re-index it.

```
node scripts/drop_index.js
node scripts/create_index.js
```

**Comment by mohitgoyal201617** [comment]
It worked. Now it treats NH-14 as NH14. 
One more thing just want to know why do we keep spaces as it  is. Means NH 14==>peliasIndexOneEdgeGram gives {'N','NH','1','14'}. Don't you think we should remove space as well becaure use  may type extra spaces while searching.

**Comment by orangejulius** [comment]
Hey @mohitgoyal201617,
First, I edited your comment above a little bit to make the formatting of code better. Let me know if i got it wrong, and I would definitely suggest using similar formatting in the future :)

I'm glad adding handling for the `-` character worked for you. @missinglink, do think we should add this filtering so that `NH-14` and `NH14` housenumbers both get indexed as `NH14`?

@mohitgoyal201617 Your changes as posted in your comment are (obviously) a bit too specific to be merged, but we would gladly help you create a pull request against [pelias/schema](https://github.com/pelias/schema/) that works across more cases. We can also help you figure out why the scoring isn't as you expect. Our [gitter](https://gitter.im/pelias/pelias) chat room might be a better place than Github comments, let me know.

**Comment by missinglink** [comment]
hi @mohitgoyal201617 I'm assuming you're referring to "National Highway No. 14 Route: Radhanpur to Beawar" (India)?

there are two ways of handling it, you can either `split` or `combine`:
```
split:
[ "NH-14" ] -> [ "NH", "14" ]

combine:
[ "NH-14" ] -> [ "NH14" ]
```

it looks like you're using `combine` which I think is correct for this situation.

Looking at your replacement function it doesn't seem to be correct, I'm surprised it's working for you.

```
"specialChar" : {
  "type":"pattern_replace",
  "pattern":"NH\\s*",
  "replacement":"NH"
}
```

The way I read this regex it says "anything starting with NH followed by a single whitespace character and then a single character of any value should be replaced with the text "NH".

Don't you want something like this?

```
"specialChar" : {
  "type":"pattern_replace",
  "pattern":"nh-([0-9]{2})",
  "replacement":"nh$1"
}
```

... which says "anything starting with "nh", followed by a "-" and then two digits is replaced with "nh" followed by the digits"?

**Comment by mohitgoyal201617** [comment]
yes ,, right
One more thing current addressit is not able to extract region,state
information correctly for indian address. Can you suggest some other
address normalizer which can be plugged.


On Tue, Jun 14, 2016 at 9:09 PM, Peter Johnson a.k.a. insertcoffee <
notifications@github.com> wrote:

> hi @mohitgoyal201617 <https://github.com/mohitgoyal201617> I'm assuming
> you're referring to "National Highway No. 14 Route: Radhanpur to Beawar"
> (India)?
>
> there are two ways of handling it, you can either split or combine:
>
> split:
> [ "NH-14" ] -> [ "NH", "14" ]
>
> combine:
> [ "NH-14" ] -> [ "NH14" ]
>
> it looks like you're using combine which I think is correct for this
> situation.
>
> Looking at your replacement function it doesn't seem to be correct, I'm
> surprised it's working for you.
>
> "specialChar" : {
>   "type":"pattern_replace",
>   "pattern":"NH\\s*",
>   "replacement":"NH"
> }
>
> The way I read this regex it says "anything starting with NH followed by a
> single whitespace character and then a single character of any value should
> be replaced with the text "NH".
>
> Don't you want something like this?
>
> "specialChar" : {
>   "type":"pattern_replace",
>   "pattern":"nh-([0-9]{2})",
>   "replacement":"nh$1"
> }
>
> ... which says "anything starting with "nh", followed by a "-" and then
> two digits is replaced with "nh" followed by the digits"?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pelias/api/issues/556#issuecomment-225922676>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/ASqL4R1oT7TMLpmvVvJ8AwCsqo0EvF_6ks5qLss-gaJpZM4Ivs4C>
> .
>



-- 
Mohit Goyal
09582396263

**Comment by orangejulius** [comment]
You are totally right. AddressIt is not very flexible globally. Fortunately we have work ongoing to use [libpostal](https://github.com/openvenues/libpostal), a machine learning project to do address parsing globally trained on OSM data. @trescube is working on integrating it into Pelias right now, and I believe initial versions for testing will be available soon.
